src/Analyzers/CSharp/Analyzers/UseThrowExpression/CSharpUseThrowExpressionDiagnosticAnalyzer.cs

```diff
@@ -23,7 +29,7 @@ protected override bool IsSupported(ParseOptions options)
             return csOptions.LanguageVersion >= LanguageVersion.CSharp7;
         }
 
-        protected override ISemanticFactsService GetSemanticFactsService()
-            => CSharpSemanticFactsService.Instance;
+        protected override bool IsInExpressionTree(SemanticModel semanticModel, SyntaxNode node, INamedTypeSymbol expressionTypeOpt, CancellationToken cancellationToken)
```

Comment:
We can consider moving `ISemanticFactsService` down to shared analyzer layer, but it seems `IsInExpressionTree` is the only API from that service used in the IDE analyzers, so I have instead switched to directly invoking the extension method.

Reply:
can you add this to the lit of work.  one of the primary reasons for ISyntax/SemanticFacts is so that we can take the direct questions like this out of the features and use a common mechanism.  

I'm fine with this for now, but would def like to see it cleaned back up.

Reply:
I added porting `ISemanticFactsService` to shared analyzer layer to the follow-up work list in https://github.com/dotnet/roslyn/issues/41462#issue-561271304